### PR TITLE
[1LP][RFR] Fix for test_service_rbac

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### PRT Run

{{pytest: cfme/tests/services/test_service_rbac.py --long-running }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit dc57da34fe1ae2344d6b6f96144d1bc003207016:

`cfme/tests/services/test_service_rbac.py`:
- :red_circle: Line 39:8: **F821** *undefined name 'self'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/services/test_service_rbac.py [hunk 1]
```diff
@@ -113,6 +119,22 @@ def test_service_rbac_service_catalog(appliance, role_user_group, catalog, catal
         service_request.wait_for_request()
         assert service_request.is_succeeded()
 
+    @request.addfinalizer
+    def _finalize():
+        rest_vm = appliance.rest_api.collections.vms.get(
+            name=f"%{catalog_item.prov_data['catalog']['vm_name']}%"
+        )
+        vm = appliance.collections.infra_vms.instantiate(name=rest_vm.name, provider=provider)
+        vm.delete()
```
Comment (jaryn) [inline cfme/tests/services/test_service_rbac.py]: Not sure guys will like `vm.delete`. Usually I am told to use `delete_if_exists` in the finalizers.

File: cfme/tests/services/test_service_rbac.py [hunk 2]
```diff
@@ -36,19 +36,23 @@ def new_user(appliance, group, credential):
...
+    if appliance.version > "5.11":
+        role = new_role(appliance=appliance, product_features=[(['Everything'], False),
+                            (['Everything', 'All VM and Instance Access Rules'], True)])
+    else:
+        role = new_role(appliance=appliance, product_features=[(['Everything'], False),
                             (['Everything', 'Access Rules for all Virtual Machines'], True)])
     group = new_group(appliance=appliance, role=role.name)
     user = new_user(appliance=appliance, group=group, credential=new_credential)
     yield role, user
     if user.exists:
```
Comment (john-dupuy) [inline cfme/tests/services/test_service_rbac.py]: these can all be changed to `delete_if_exists`

File: cfme/tests/services/test_service_rbac.py [hunk 3]
```diff
@@ -36,19 +36,23 @@ def new_user(appliance, group, credential):
...
-        user.delete(cancel=False)
+        user.delete()
 
     if group.exists:
-        group.delete(cancel=False)
+        group.delete()
```
Comment (john-dupuy) [inline cfme/tests/services/test_service_rbac.py]:
```suggestion
    group.delete_if_exists()
```

Comment (john-dupuy) [reply cfme/tests/services/test_service_rbac.py]: Same for `user` and `role`

File: cfme/tests/services/test_service_rbac.py (hunk 1 above)
Comment (john-dupuy) [reply cfme/tests/services/test_service_rbac.py]: Yup! I'd prefer `delete_if_exists` here

File: cfme/tests/services/test_service_rbac.py [hunk 4]
```diff
@@ -185,3 +207,19 @@ def test_service_rbac_request(appliance, role_user_group, catalog_item):
...
         order_request.wait_for_request(method='ui')
         assert order_request.is_succeeded(method='ui')
+
+    @request.addfinalizer
+    def _finalize():
+        rest_vm = appliance.rest_api.collections.vms.get(
+            name=f"%{catalog_item.prov_data['catalog']['vm_name']}%"
+        )
+        vm = appliance.collections.infra_vms.instantiate(name=rest_vm.name, provider=provider)
+        vm.delete()
```
Comment (john-dupuy) [inline cfme/tests/services/test_service_rbac.py]:
```suggestion
        vm.delete_if_exists()
```

File: cfme/tests/services/test_service_rbac.py [hunk 5]
```diff
@@ -36,19 +36,19 @@ def new_user(appliance, group, credential):
 
 @pytest.yield_fixture(scope='module')
 def role_user_group(appliance, new_credential):
-    role = new_role(appliance=appliance, product_features=[(['Everything'], False),
+    if appliance.version > "5.11":
+        role = new_role(appliance=appliance, product_features=[(['Everything'], False),
+                            (['Everything', 'All VM and Instance Access Rules'], True)])
+    else:
+        role = new_role(appliance=appliance, product_features=[(['Everything'], False),
```
Comment (digitronik) [inline cfme/tests/services/test_service_rbac.py]: Optional: the `preoduct_features` are different, I would just mauculate that.
```
vm_access_rule = 'All VM and Instance Access Rules' if appliance.version > "5.11" else 'Access Rules for all Virtual Machines'
role = new_role(appliance=appliance, product_features=[(['Everything'], False),
                            (['Everything', vm_access_rule], True)])
```

File: cfme/tests/services/test_service_rbac.py [hunk 6]
```diff
@@ -36,19 +36,19 @@ def new_user(appliance, group, credential):
...
-        user.delete(cancel=False)
 
-    if group.exists:
-        group.delete(cancel=False)
-
-    if role.exists:
-        role.delete(cancel=False)
+    user.delete_if_exists()
+    group.delete_if_exists()
+    role.delete_if_exists()
```
Comment (digitronik) [inline cfme/tests/services/test_service_rbac.py]: :+1:

File: cfme/tests/services/test_service_rbac.py [hunk 7]
```diff
@@ -113,6 +115,22 @@ def test_service_rbac_service_catalog(appliance, role_user_group, catalog, catal
         service_request.wait_for_request()
         assert service_request.is_succeeded()
 
+    @request.addfinalizer
+    def _finalize():
```
Comment (digitronik) [inline cfme/tests/services/test_service_rbac.py]: Thanks for this cleanup :+1:
If user has access to delete request and VM; I think better to move this finalizer after `order`and we don't need to instantiate request again; order method will return it for you.

**Pull request merged**